src/compiler/checker.ts

```diff
@@ -40778,6 +40778,44 @@ namespace ts {
                 // otherwise it will conflict with some local declaration). Note that in addition to normal flags we include matching SymbolFlags.Export*
                 // in order to prevent collisions with declarations that were exported from the current module (they still contribute to local names).
                 symbol = getMergedSymbol(symbol.exportSymbol || symbol);
+
+                // A type-only import/export will already have a grammar error in a JS file, so no need to issue more errors within
+                if (isInJSFile(node) && !(target.flags & SymbolFlags.Value) && !isTypeOnlyImportOrExportDeclaration(node)) {
+                    const errorNode =
+                        isImportOrExportSpecifier(node) ? node.propertyName || node.name :
+                        isNamedDeclaration(node) ? node.name :
+                        node;
+
+                    Debug.assert(node.kind !== SyntaxKind.NamespaceExport);
+                    const isExport = node.kind === SyntaxKind.ExportSpecifier;
+                    if (isExport) {

```

Comment:
```suggestion
                    // Give a specialized message depending on whether we're exporting or importing
                    if (node.kind === SyntaxKind.ExportSpecifier) {
```

You don't use `isExport` anywhere else. Might as well just add a comment above

Reply:
On the other hand, I kind of like this style too. It's your call.